src/main.rs

```diff
@@ -1,110 +1,24 @@
-use my_project::interpreter::Interpreter;
-
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    println!("Hello, world!");
-
-    let mut interpreter = Interpreter::new();
-    interpreter.run("var x 10 print x")?;
+use std::{env, fs};
 
-    let mut interpreter1 = Interpreter::new();
-    interpreter1.run("var x 10 if x == 10 print x")?; // var x 10 if x 10 var y 20 else var y 30 end print y
+use my_project::interpreter::Interpreter;
 
-    let mut interpreter2 = Interpreter::new();
-    interpreter2.run("var x 30 var y 10 add x y print x end")?;
+static ERROR_MISSING_INPUT_FILE: &str = "Missing input file";
+static ERROR_FILE_READ: &str = "Could not read file";
+static ERROR_BAD_EXTENSION: &str = "Source files must have the .nek file extension";
 
-    let mut interpreter3 = Interpreter::new();
-    let source = "var x 30 
-                  var y 10 
-                  var z 40 
-                  add x y 
-                  print x
-                  add x z
-                  print x
-                  end";
-    interpreter3.run(source)?;
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args: Vec<String> = env::args().collect();
 
-    let mut interpreter4 = Interpreter::new();
-    let source1 = "
-        var x 0
-        var y 1
-        loop x < 5
-            print x
-            add x y
-            end
-        end
-    ";
-    interpreter4.run(source1)?;
+    let source_file = args.iter().skip(1).next().expect(ERROR_MISSING_INPUT_FILE);
+    
+    if ! source_file.ends_with(".nek"){
+        panic!("{}", ERROR_BAD_EXTENSION);
+    }
 
-    let mut interpreter5 = Interpreter::new();
-    let source2 = "
-        var x 10
-        var y 5
-        var z 1
-        if x > 5
-          add x y
-        end
-        loop x < 20
-            print x
-            add x z
-            end
-        end
-    ";
-    interpreter5.run(source2)?;
+    let code = fs::read_to_string(source_file).expect(&format!("{} {}", ERROR_FILE_READ, source_file));
```

Comment:
`.map_err(|e| format!("{} {}: {}", ERROR_FILE_READ, source_file, e))?;` instead of panic